pkg/connector/vault.go

```diff
@@ -123,18 +137,34 @@ func (g *vaultResourceType) Entitlements(ctx context.Context, resource *v2.Resou
 	}
 
 	memberOptions := PopulateOptions(resource.DisplayName, memberEntitlement, resource.Id.ResourceType)
-	memberEntitlement := ent.NewAssignmentEntitlement(resource, memberEntitlement, memberOptions...)
-	rv = append(rv, memberEntitlement)
+	membetEnt := ent.NewAssignmentEntitlement(resource, memberEntitlement, memberOptions...)
```

Comment:
When we've done similar filtering in connectors, we like to extract this logic into a helper:
```go
func shouldIncludeVault(limitVaultPermissions mapset.Set[string]) bool {
	// If there are no explicit limits, then always include the vault.
	if limitVaultPermissions == nil {
		return true
	}
	// If there are explicit limits, only include this vault if it was selected.
	return limitVaultPermissions.Contains(memberEntitlement)
}
```
```go
if shouldIncludeVault(g.limitVaultPermissions) {
    rv = append(rv, memberEnt)
}
```